### ISSUE TITLE
Starting two datacenter instances in one server

### ISSUE THREAD

**Issue by Mariius** [hashicorp/consul#4569]
Hello,

Currently I am implementing Consul in order to use it as Health-checking tool.

Here is my issue:
We have Production environment and Dev environemt. Both of these environments have their services.

What I want to do is have one Consul server with two datacenters called "prod" and "dev". Agents would report healhchecks to "prod" or "dev" depending on service.

The problem is that one Consul server can represent one datacenter. And I am not able to run two instances of Consul in one physical server because the ports are already occupied by one Consul instance.

Is it possible to somehow in one physical server start Consul server so that it has two datacenters inside it?

**Comment by kyhavlov** [comment]
It's not possible to have one server be part of two datacenters (and strongly not recommended to have both prod/dev share a cluster), but it *is* possible to run multiple instances of Consul on the same machine by changing the ports in the agent configuration: https://www.consul.io/docs/agent/options.html#ports

Even in that case, it's probably not a great idea to have dev and prod sharing machines and you'd want to separate them if possible. You could also use [Service ACL permissions](https://www.consul.io/docs/guides/acl.html#service-rules) to prevent users in one segment from reading/writing to the other, but again we really recommend not sharing a cluster between dev/prod like that.

Hope this helps!

**Comment by Mariius** [comment]
Thank you for your time. I was wondering isn't it possible to start two docker consul server containers in the  same machine with different datacenters and make them communicate with each other?

**Comment by Mariius** [comment]
I would like to edit my question a bit. I think I am totally lost and am not able to understand how Consul works.

For testing purposes I started one server instance on one machine with datacenter "prod". I have started another consul instance on another machine with datacenter "dev".
After "prod" is started, i try to join "dev" and it outputs "Failed to join 172.x.x.x: Member 'Consul Dev Server' part of wrong datacenter 'dev'"

So, how can I create two datacenters and have one dashboard for them?
